aws/aws-how-to/kubernetes/deploy-ubuntu-pro-fips-cluster.rst

```diff
@@ -0,0 +1,239 @@
+Deploy an EKS FIPS cluster with Ubuntu Pro
+==========================================
+
+This guide will walk you through the steps needed to get an EKS cluster running Ubuntu on FIPS mode to achieve FedRamp compliance. 
+
+The process requires you to create your custom EKS FIPS AMI based on `recently released Ubuntu Pro for EKS`_ AMI using Packer, to then deploy it using eksctl. We will also show how to deploy a demo container that can be used to test and peek inside the cluster.
+
+Prerequisites
+~~~~~~~~~~~~~
+
+- An AWS account
+- AWS CLI installed and configured
+- An access and secret key from IAM
+- ``Packer`` version 1.8.1 or newer installed. Check the instructions to `install packer`_
+- ``eksctl`` version 1.7.7 or newer installed. Check the instructions to `install eksctl`_
+- ``Kubectl`` installed. Install it with ``sudo snap install kubectl --classic``
+
+Step 1. Build your custom EKS AMI using ‘Ubuntu-Pro for EKS’
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create a json file (say eks-fips.json) for use with Packer: 
+
+
+..  code-block:: json
+
+    {
+        "variables": {
+            "aws_access_key": "<<YOUR_ACCESS_KEY>>",
+            "aws_secret_key": "<<YOUR_SECRET_KEY>>",
+            "eks_ver": "1.29"
+        },
+         "builders": [
+          {
+            "type": "amazon-ebs",
+            "access_key": "{{user `aws_access_key`}}",
+            "secret_key": "{{user `aws_secret_key`}}",
+            "region": "us-east-1",
+            "instance_type": "t2.micro",
+            "ami_name": "eks{{user `eks_ver`}}-fips-ubuntu22.04-{{timestamp}}",
+            "source_ami_filter": {
+                "filters": {
+                    "virtualization-type": "hvm",
+                    "name": "ubuntu-eks-pro/k8s_{{user `eks_ver`}}/images/hvm-ssd/ubuntu-jammy-22.04-amd64-server-*",
+                    "root-device-type": "ebs"
+                },
+            "owners": ["099720109477"],
+            "most_recent": true
+          },
+          "ssh_username": "ubuntu"
+          }
+        ],
+        "provisioners": [
+          {
+            "type": "shell",
+            "inline": [
+              "cloud-init status --wait",
+              "sudo apt-get update && sudo apt-get upgrade -y --with-new-pkgs",
+              "sudo pro enable fips-updates --assume-yes"
```

Comment:
maybe it would be good to add a note on this page outlining the main difference between FIPS and FIPS updates and point to a reference doc.

Reply:
Since Carlos has a customer requesting this exact documentation, I am going to publish it as it is for now and will add the this update in a separate PR.